### PR TITLE
feat(install): register hooks for manual Claude installs

### LINKED ISSUE #2982: `--enable-hooks` silently no-ops on `--target claude`, and install-state reports success
# `--enable-hooks` silently no-ops on `--target claude`, and install-state reports success

**ECC version:** 2.2.1
**Harness:** Claude Code 2.1.258
**OS:** Linux (Cinnamon desktop, kernel 7.0.0-30-generic)
**Node:** v24.19.0
**Install method:** git clone + `node scripts/install-apply.js` (no plugin, no npx)

## Summary

Installing with `--target claude --enable-hooks` copies `hooks.json` into
`~/.claude/hooks/hooks.json` but never registers the hooks anywhere Claude Code
reads them. No hook ever fires. `install-state.json` nonetheless records
`hookConsent: "enabled"` and the installer exits 0, so the failure is silent.

Hook activation appears to be entirely plugin-mediated
(`scripts/lib/claude-plugin-setup.js` drives it through
`pluginConfigs[<plugin>].options.hooks_enabled`). The `claude` home target has
no equivalent activation path, but still accepts `--enable-hooks` and reports
it as applied.

## Reproduction

```bash
git clone https://github.com/affaan-m/everything-claude-code.git ecc
cd ecc && npm install
node scripts/install-apply.js --target claude --profile full --enable-hooks
```

Then run the project's own verification step from `TROUBLESHOOTING.md:173`:

```bash
grep -A 10 '"hooks"' ~/.claude/settings.json
```

## Expected

Hooks registered so Claude Code loads them; or the installer refuses
`--enable-hooks` for a target that cannot activate them.

## Actual

```
$ grep -A 10 '"hooks"' ~/.claude/settings.json
(no match)

$ ls ~/.claude/plugins/
ls: cannot access '/home/.../.claude/plugins/': No such file or directory

$ python3 -c "import json;print(json.load(open('$HOME/.claude/ecc/install-state.json'))['request']['hookConsent'])"
enabled
```

The hook scripts themselves are fine. Invoked directly they behave correctly,
including exit codes and the `hookSpecificOutput` protocol:

```
$ echo '{"tool_name":"Bash","tool_input":{"command":"git commit --no-verify -m x"}}' \
    | node ~/.claude/scripts/hooks/block-no-verify.js
BLOCKED: --no-verify flag is not allowed with git commit. Git hooks must not be bypassed.
$ echo $?
2
```

They are simply never invoked. Manually merging `~/.claude/hooks/hooks.json`
into the `hooks` key of `~/.claude/settings.json` makes all 23 hook commands
fire correctly, which confirms the only missing step is registration.

## Related issues found while diagnosing

**1. The installer performs an action its own docs forbid.**
`hooks/README.md:24` states:

> For Claude Code manual installs, do not paste the raw repo `hooks.json` into
> `~/.claude/settings.json` or copy it directly into `~/.claude/hooks/hooks.json`.

`scripts/lib/install/apply.js:311-463` writes exactly that destination path
(`path.join(plan.targetRoot, 'hooks', 'hooks.json')`) for the `claude` target.

**2. `--dry-run` is not a faithful preview.**
The dry-run plan for `--profile full --enable-hooks` reports 1049 operations,
all of kind `copy-file`, with no `settings.json` destination. The apply path
then calls `writeClaudeCommitAttributionPreference()` at
`scripts/lib/install/apply.js:474`, adding `includeCoAuthoredBy: false` to
`~/.claude/settings.json`. That mutation is never surfaced in the plan.

**3. `harness-audit.js` cannot see a home-target install.**
After a successful `--target claude` install of 68 agents, 94 commands, 286
skills and 122 rules, the bundled audit still reports:

```
Harness Audit (repo, consumer): 12/39
1) [Tool Coverage] Install the ECC plugin for this user or project before
   auditing project-specific harness quality.
   (~/.claude/plugins/ecc/ (legacy everything-claude-code paths also supported))
```

The audit only looks for the plugin layout, so the officially supported home
install is invisible to it.

## Suggested fixes

In rough priority order:

1. Have the `claude` home adapter merge `hooks.json` into the `hooks` key of
   `~/.claude/settings.json` during apply — the code already writes that file
   for commit attribution, so the write path exists.
2. Failing that, make `--enable-hooks` a hard error on targets that cannot
   activate hooks, rather than recording `hookConsent: "enabled"`.
3. Declare the `settings.json` mutation as an operation in the dry-run plan.
4. Teach `harness-audit.js` to detect home-target installs via
   `~/.claude/ecc/install-state.json`.

Happy to test a patch against this setup.

---

*Investigated and written by a bot, posted on behalf of eloopj.*

### PR DESCRIPTION
**Pull request by wellkilo**
## What Changed

- add a dedicated `update-claude-settings` install operation for manual `claude` and `claude-project` targets
- register all managed hooks in the canonical Claude `settings.json` by stable event/id ownership while preserving user settings and hooks
- support idempotent upgrades, `--no-hooks` removal, doctor, repair, and uninstall without restoring whole-file snapshots
- fail closed on malformed settings, same-id conflicts, drift, symlink destinations, and unsafe install-state metadata
- serialize ECC writers with a settings lock and write settings atomically with private permissions
- validate stable, globally unique matcher IDs and document the manual-install behavior

## Why This Change

Manual installs accepted `--enable-hooks`, copied `hooks/hooks.json` to a location Claude Code does not load, and still recorded hooks as enabled. This makes the planned operation, applied configuration, and lifecycle state agree while keeping plugin installs unchanged.

Closes #2982

## Testing Done

- [x] Manual testing completed
- [x] Automated tests pass in GitHub CI
- [x] Edge cases considered and tested

Focused verification passes with 177/177 tests across settings ownership, installer execution, consent, install state, doctor/repair/uninstall, CLI apply, and docs. Claude plugin compatibility tests pass 45/45. The full GitHub CI matrix passes 45/45 across Linux, macOS, and Windows with Node 18/20/22 and npm/pnpm/yarn/Bun, including coverage, lint, security, component validation, and packed-install lifecycle checks. The Claude settings modules exceed the 80% focused coverage gate (87.52% statements/lines, 81.85% branches, 94.28% functions).

## Type of Change

- [ ] `fix:` Bug fix
- [x] `feat:` New feature
- [ ] `refactor:` Code refactoring
- [ ] `docs:` Documentation
- [ ] `test:` Tests
- [ ] `chore:` Maintenance/tooling
- [ ] `ci:` CI/CD changes

## Security & Quality Checklist

- [x] No secrets or API keys committed (ghp_, sk-, AKIA, xoxb, xoxp patterns checked)
- [x] JSON files validate cleanly
- [ ] Shell scripts pass shellcheck (if applicable)
- [ ] Pre-commit hooks pass locally (if configured)
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## If you changed dependencies or `package.json` (`bin` / `files` / deps)

- [ ] Ran `yarn install --mode=update-lockfile` and committed the `yarn.lock` change. CI runs Yarn in hardened mode on public PRs and fails if the lockfile would be modified, so an out of date `yarn.lock` breaks the build even when nothing else is wrong.

Not applicable: no dependency or package metadata changes.

## If you added a skill, command, agent, hook, or CLI tool

- [ ] Registered in `package.json` (`bin` and `files`), `manifests/install-components.json`, `manifests/install-modules.json`, and `agent.yaml`
- [ ] Regenerated the catalog (`npm run catalog:sync`) and command registry (`npm run command-registry:write`)
- [ ] Updated the docs tables it belongs in (`README.md`, `COMMANDS-QUICK-REF.md`, `docs/COMMAND-AGENT-MAP.md`)
- [ ] If it ships a new script path, added it to the publish surface allowlist (`tests/scripts/npm-publish-surface.test.js`)
- [ ] Cross-harness surfaces updated if applicable (for Codex, `.agents/skills/<name>/` plus `agents/openai.yaml`; the Codex frontmatter validator allows only `name`, `description`, `metadata`, `license`, `allowed-tools`, so drop keys like `version` from that copy)
- [ ] Full gauntlet passes locally (`npm test`)

Not applicable: this updates registration for the existing hook runtime and does not add a catalog surface.

## Documentation

- [x] Updated relevant documentation
- [x] Added comments for complex logic
- [x] README updated (if needed)

### CODE REVIEW

File: scripts/lib/install/claude-settings.js [hunk 1]
```diff
@@ -0,0 +1,750 @@
+'use strict';
+
+const crypto = require('crypto');
+const fs = require('fs');
+const path = require('path');
+const { isDeepStrictEqual } = require('util');
+const { writeFileAtomic } = require('../atomic-write');
```
[the remark this answers is not in the document]
Comment (wellkilo) [reply scripts/lib/install/claude-settings.js:6]: Addressed in f59cfd57 by extracting the independent lock and stale-lock recovery logic into `claude-settings-lock.js` while preserving the existing `claude-settings.js` API. The remaining validation, materialization, and ownership transforms intentionally stay together because they share the same hook invariants; the main helper is now 672 lines, below the repository 800-line maximum.
